src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -271,6 +274,42 @@ public static void daemonInitialization(Supplier<Config> config) throws Configur
         AuthConfig.applyAuth();
     }
 
+    public static void unsafeDaemonInitialization(Supplier<Config> config) throws ConfigurationException
+    {
+        if (toolInitialized)
+            throw new AssertionError("toolInitialization() already called");
+        if (clientInitialized)
+            throw new AssertionError("clientInitialization() already called");
```

Comment:
nit: We could de-dup the 3 places where these are called w/ something like...

```
assertNotToolInitialized();
assertNotClientInitialized();
```

Not much ROI, but up to you.

Reply:
added `assertNotDaemonInitialized` as well